src/StreamJsonRpc/Reflection/MessageFormatterRpcMarshaledContextTracker.cs

```diff
@@ -157,20 +160,34 @@ internal MarshalToken GetToken(IRpcMarshaledContext<object> marshaledContext)
 
         long handle = this.nextUniqueHandle++;
 
+        Type actualInterface = declaredType;
+        int? subTypeCode = null;
+        foreach (RpcMarshalableKnownSubtypeAttribute attribute in declaredType.GetCustomAttributes(typeof(RpcMarshalableKnownSubtypeAttribute), inherit: false))
```

Comment:
Is this amount of reflection of upon serialization and deserialization?

Reply:
And while we're caching the initial result, we should validate it:
1. each ordinal value is used at most once.
1. each subtype actually derives from the interface the attribute is applied to.
1. each subtype is used at most once.